### PR TITLE
Feat/dyanmicdata

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview

Successfully implemented fetching of dynamic data for users. Implementation of serverless function to handle requests to Twitter API.

Features:
1. Implementation of serverless functions hosted on Vercel
2. Update to logic to request data and handle accordingly

### CODE REVIEW

File: src/utils/twitterAPI/useTwitter.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+
+export const fetchTwitterUser = async (username, dispatch) => {
+  try {
+    const res = await axios.get(
+      `https://twitter-api-fetch-userdata.netlify.app/api/fetchUserData?username=${username}`
+    );
+    if (res)
```
Comment (abhij1607) [inline src/utils/twitterAPI/useTwitter.jsx:8]: since this is in try block, so if condition wasn't needed

Comment (Hamdrive) [reply src/utils/twitterAPI/useTwitter.jsx:8]: Actually had to implement check if res contained data or not, have updated it in the net PR

File: src/context/auth-context.jsx [hunk 2]
```diff
@@ -18,23 +18,14 @@ const AuthProvider = ({ children }) => {
...
-
   const signupUser = async (name, email, password) => {
     try {
       setLoading(true);
       const res = await createUserWithEmailAndPassword(auth, email, password);
       const updateUserProfile = await updateProfile(auth.currentUser, {
         displayName: name,
       });
-      localStorage.setItem("userID", JSON.stringify(res.user.uid) )
+      localStorage.setItem("userID", JSON.stringify(res.user.uid));
```
Comment (prachi-sahani) [inline src/context/auth-context.jsx:28]: Since `uid` is itself is a string, you need not `stringify` it

**Review by prachi-sahani — COMMENTED** [review]
Code is clean.
The changes are good to merge.

**Review by abhij1607 — COMMENTED** [review]
Neat and Clean code
Good in Logic implementaion and error handling
Can be merged

Suggestion
- clear console
- I think, In **try{}** - _**if condition** won't be required to check wheater data is present_

File: src/context/auth-context.jsx (hunk 2 above)
Comment (Hamdrive) [reply src/context/auth-context.jsx:28]: Noted, have made the changes in the next PR

**Pull request merged**